types/meteor/test/meteor-mongo-tests.ts

```diff
@@ -0,0 +1,499 @@
+    },
+    remove: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType { foo: string; name: string; }
+        doc;
+        return true;
+    },
+    insert: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType { foo: string; name: string; }
+        doc;
+        return true;
+    },
+    transform: (doc: DBUser) => ({ foo: 'foo', name: doc.name }),
+});
+
+Users.allow({
+    update: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+    remove: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+    insert: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+});
+
+Users.allow({
+    update: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+    remove: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+    insert: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType DBUser
+        doc;
+        return true;
+    },
+    transform: null,
+});
+
+Users.allow({
+    update: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType { foo: string; name: string; }
+        doc;
+        return true;
+    },
+    remove: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType { foo: string; name: string; }
+        doc;
+        return true;
+    },
+    insert: (userId, doc) => {
+        // $ExpectType string
+        userId;
+        // $ExpectType { foo: string; name: string; }
+        doc;
+        return true;
+    },
+    transform: (doc: DBUser) => ({ foo: 'foo', name: doc.name }),
+});
+
+Users.find({}).observe({
+    changed: (newDoc, oldDoc) => {
+        // $ExpectType DBUser
+        newDoc;
+        // $ExpectType DBUser
+        oldDoc;
+        return false;
+    },
+});
+
+Users.find({}, { transform: (doc: DBUser) => 'null' }).observe({
+    changed: (newDoc, oldDoc) => {
+        // $ExpectType string
+        newDoc;
+        // $ExpectType string
+        oldDoc;
+        return false;
+    },
+});
+
+Users.find({}, { transform: null }).observe({
+    changed: (newDoc, oldDoc) => {
+        // $ExpectType DBUser
+        newDoc;
+        // $ExpectType DBUser
+        oldDoc;
+        return false;
+    },
+});
+
+// $ExpectType DBUser | undefined
+Users.findOne({});
+// $ExpectType DBUser | undefined
+Users.findOne({}, {});
+// $ExpectType DBUser | undefined
+Users.findOne({}, { transform: null });
+// $ExpectType string | undefined
+Users.findOne({}, { transform: doc => 'null' });
```

Comment:
That won't work as [`transform` must return an object](https://github.com/meteor/meteor/blob/d9db4f52f2ea6d706a25156768ea42e1fbb8f599/packages/minimongo/local_collection.js#L779-L785).

Reply:
Thanks, I missed that.

Reply:
Should I take into account that under the hood it is assigning { _id: string } to the returned object?

Reply:
I wouldn't say so. There's only one `_id` related limitation - you cannot change it. Returning an object without `_id` is fine as well. Also, `_id` is not always a string. I think it'd be enough to make it return an object, maybe `{}` is enough.